app/views/contexts/calculated_summary_context.py

```diff
@@ -162,21 +203,70 @@ def _get_formatted_total(self, groups, current_location, calculation_operator):
                     values_to_calculate.append(answer_value)
 
         calculated_total = calculation_operator(values_to_calculate)
+
+        return self._format_total(answer_format, calculated_total)
+
+    def _get_formatted_total_with_rule_evaluator(
+        self, groups: list, current_location: Location, calculation: ImmutableDict
+    ) -> str:
+        answer_format: Mapping = {"type": None}
+        for group in groups:
+            for block in group["blocks"]:
+                question = choose_question_to_display(
+                    block,
+                    self._schema,
+                    self._metadata,
+                    self._response_metadata,
+                    self._answer_store,
+                    self._list_store,
+                    current_location=current_location,
+                )
+                for answer in question["answers"]:
+                    if not answer_format["type"]:
+                        answer_format = {
+                            "type": answer["type"],
+                            "unit": answer.get("unit"),
+                            "unit_length": answer.get("unit_length"),
+                            "currency": answer.get("currency"),
+                        }
```

Comment:
Avoid duplicating this, this is the main bit, so ideally 1 source. I suggest a single method and you can just check if it is legacy or new using `if "calculation_type" in calculation:` I suggest keep the existing method, but the end just either call RE or the current `sum()`